crates/driver/src/domain/mempools.rs

```diff
@@ -123,24 +128,32 @@ impl Mempools {
                     TxStatus::Pending => {
                         // Check if the current block reached the submission deadline block number
                         if block.number >= submission_deadline {
+                            let cancellation_tx_hash = self
+                                .cancel(mempool, settlement.gas.price, solver)
+                                .await
+                                .context("cancellation tx due to deadline failed")?;
                             tracing::info!(
-                                ?hash,
+                                settle_tx_hash = ?hash,
```

Comment:
We are now skipping the log if `self.cancel` returns error. Hope this is intended by this PR?

Reply:
Yes, we do. In case of cancellation fails, we have an enriched context error to understand which path that was.